validation/src/test/scala/hmda/validation/rules/lar/quality/Q022Spec.scala

```diff
@@ -0,0 +1,38 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.parser.fi.lar.LarGenerators
+import hmda.validation.dsl.Success
+import hmda.validation.dsl.Failure
+import hmda.validation.rules.lar.BadValueUtils
+import org.scalacheck.Gen
+import org.scalatest.{ MustMatchers, PropSpec }
+import org.scalatest.prop.PropertyChecks
+
+class Q022Spec extends PropSpec with PropertyChecks with MustMatchers with LarGenerators with BadValueUtils {
+  // Cases meeting preconditions
+  property("Passes if activity year is within two years after application date") {
+    forAll(larGen, Gen.choose(0, 2)) { (lar, x) =>
+      whenever(lar.loan.applicationDate != "NA") {
+        val applicationYear = lar.loan.applicationDate.substring(0, 4).toInt
+        Q022(lar, applicationYear + x) mustBe a[Success]
```

Comment:
I'd also like to see tests for the `apply` method that returns a `Future[Result]`. `S013Spec` seems to show a simple strategy for doing that.